File: atividade-18/atividade.py
class AFD:
    def __init__(self):
        
        self.state = 'q00'  

    def transition(self, char):
        if self.state == 'q00':
            if char == '0':
                self.state = 'q10'
            elif char == '1':
                self.state = 'q01'
        elif self.state == 'q01':
            if char == '0':
                self.state = 'q11'
            elif char == '1':
                self.state = 'q00'
        elif self.state == 'q10':
            if char == '0':
                self.state = 'q00'
            elif char == '1':
                self.state = 'q11'
        elif self.state == 'q11':
            if char == '0':
                self.state = 'q01'
            elif char == '1':
                self.state = 'q10'

    def is_accepted(self, string):
        self.state = 'q00'  
        for char in string:
            self.transition(char)
        return self.state == 'q11'  

if __name__ == '__main__':
  afd = AFD()
  print(afd.is_accepted("01"))  
  print(afd.is_accepted("1100"))  
  print(afd.is_accepted("10"))  
  print(afd.is_accepted("0000"))  
  print(afd.is_accepted("1111"))  

File: atividade-18/atividade.go
package main

import "fmt"

type AFD struct {
	state string
}

func (afd *AFD) transition(char rune) {
	switch afd.state {
	case "q00":
		if char == '0' {
			afd.state = "q10"
		} else if char == '1' {
			afd.state = "q01"
		}
	case "q01":
		if char == '0' {
			afd.state = "q11"
		} else if char == '1' {
			afd.state = "q00"
		}
	case "q10":
		if char == '0' {
			afd.state = "q00"
		} else if char == '1' {
			afd.state = "q11"
		}
	case "q11":
		if char == '0' {
			afd.state = "q01"
		} else if char == '1' {
			afd.state = "q10"
		}
	}
}

func (afd *AFD) isAccepted(input string) bool {
	afd.state = "q00" 
	for _, char := range input {
		afd.transition(char)
	}
	return afd.state == "q11" 
}

func main() {
	afd := AFD{}
	fmt.Println(afd.isAccepted("01"))   
	fmt.Println(afd.isAccepted("1100")) 
	fmt.Println(afd.isAccepted("10"))   
	fmt.Println(afd.isAccepted("0000")) 
	fmt.Println(afd.isAccepted("1111")) 
}
